src/background/Wallet/Wallet.ts

```diff
@@ -1017,14 +1018,17 @@ export class Wallet {
   }
 
   async switchChainForOrigin({
-    params: { chain, origin },
+    params: { evmChain, solanaChain, origin },
     context,
-  }: WalletMethodParams<{ chain: string; origin: string }>) {
+  }: WalletMethodParams<
+    AtLeastOneOf<{ evmChain: string; solanaChain: string }> & { origin: string }
```

Comment:
Aren't this the same type as on the line 1031? 
`Parameters<Wallet['setChainForOrigin']>[0])`

Reply:
ts doesn't recognize that the `either-or` constraint is kept